### PR TITLE
Start independent services in kube-compose defined in docker-compose.yml

### PR DESCRIPTION
**Pull request by maheshrayas**
@xUnholy @jbrekelmans  PTAL

### CODE REVIEW

File: pkg/up/up.go [hunk 1]
```diff
@@ -337,7 +380,9 @@ func (u *upRunner) createServicesAndGetPodHostAliases() ([]v1.HostAlias, error)
 			u.initResourceObjectMeta(&service.ObjectMeta, app.nameEncoded, app.name)
 			_, err := u.k8sServiceClient.Create(service)
 			if err != nil {
-				return nil, err
+				if err.Error() != errors.New("services \""+service.Name+"\" already exists").Error() {
```
Comment (jbrekelmans) [inline pkg/up/up.go]: For a string `s`, `errors.New(s).Error()` can be simplified to `s`

Comment (jbrekelmans) [reply pkg/up/up.go]: There is probably a reason why serviceName is double quoted (special character may be backslash escaped), so this method of error parsing may break for some service names. Is there an official way to do error checking for k8s client-go? (e.g. cast to specific error and check an error code).

File: pkg/up/up.go [hunk 2]
```diff
@@ -680,7 +727,7 @@ func (u *upRunner) run() error {
 			break
 		}
 	}
-	fmt.Printf("pods ready (%d/%d)\n", len(u.apps), len(u.apps))
+	fmt.Printf("pods ready (%d/%d)\n", len(u.appsThatNeedToBeReady), len(u.appsThatNeedToBeReady))
```
Comment (jbrekelmans) [inline pkg/up/up.go]: I believe there is another log statement which prints a similar line, but that one has not been updated.

File: pkg/up/up.go (hunk 1 above)
Comment (jbrekelmans) [reply pkg/up/up.go]: We should change the log message if the service already exists ("service already exists")

File: pkg/up/up.go [hunk 3]
```diff
@@ -97,8 +98,47 @@ func (u *upRunner) initKubernetesClientset() error {
...
+func contains(arr []string, str string) bool {
+	for _, a := range arr {
+		if a == str {
+			return true
+		}
+	}
+	return false
+}
+
+func (u *upRunner) podsToBeCreated() error {
```
[the remark this answers is not in the document]
Comment (maheshrayas) [reply pkg/up/up.go]: Fixed

File: pkg/up/up.go [hunk 4]
```diff
@@ -97,8 +98,47 @@ func (u *upRunner) initKubernetesClientset() error {
 	return nil
 }
 
+func contains(arr []string, str string) bool {
```
[the remark this answers is not in the document]
Comment (maheshrayas) [reply pkg/up/up.go]: Moved to utils.go

File: pkg/up/up.go (hunk 1 above)
Comment (maheshrayas) [reply pkg/up/up.go]: k8s apimachinery handles the "already existing error". Implemented the same in this case.

File: pkg/up/up.go (hunk 2 above)
Comment (maheshrayas) [reply pkg/up/up.go]: IMO the similar check of Services cannot be applied here. As we are trying to create a pod and pod returns error message immediately. Further down in the code, we try to receive the message from the pod status channel and channel never sends any message.

File: pkg/up/up.go [hunk 5]
```diff
@@ -336,10 +373,13 @@ func (u *upRunner) createServicesAndGetPodHostAliases() ([]v1.HostAlias, error)
 			}
 			u.initResourceObjectMeta(&service.ObjectMeta, app.nameEncoded, app.name)
 			_, err := u.k8sServiceClient.Create(service)
-			if err != nil {
-				return nil, err
+			if k8sError.IsAlreadyExists(err) {
+				fmt.Printf("service %s already exists\n", service.Name)
```
Comment (jbrekelmans) [inline pkg/up/up.go]: Make log format consistent with below log.

Comment (maheshrayas) [reply pkg/up/up.go]: @jbrekelmans  Fixed. PTAL

**Pull request merged**